docs/api.rst

```diff
@@ -6,7 +6,38 @@
 
 
 LIBBPF API
-==================
+==========
+
+Error Handling
+--------------
+
+API functions in libbpf can return errors in one of two ways. 
```

Comment:
Can you please add a note that all this applies to "libbpf 1.0" mode which can be set with `libbpf_set_strict_mode(LIBBPF_STRICT_DIRECT_ERRS | LIBBPF_STRICT_CLEAN_PTRS);`?

Reply:
Can you elaborate on how this works? Will this be default starting at 1.0? Is it not on by default now?

Reply:
This will be the only behavior in 1.0, yes. It's not on by default right now, you have to "opt-in" for the transition period before 1.0